Added Language Module

Entree3k opened the pull request:
Added a Language Module so you can see what Language a movie is in. Also has ability to skip dual audio videos so you can just see files that have one audio track

Entree3k committed:
Added Language.py

Language.py is for detecting the Language of an audio track

Entree3k committed:
Added Languge.py

amadib commented on modules/Language.py:
```diff
@@ -0,0 +1,139 @@
+            return None
+
+        # Mapping of common language codes to full names in alphabetical order
+        language_mapping = {
```
Would it be possible to extend this to also support [language tags](https://www.rfc-editor.org/rfc/rfc9110.html#name-language-tags)?

Entree3k replied:
Like mi, en, da, etc? or what tag format?

amadib replied:
Also referred to as `locales`. Here's another [reference](https://en.wikipedia.org/wiki/Language_localisation). 

Personally example much of my library includes Arabic that's either dubbed into both classical/modern standard arabic (`ar`) or egyptian arabic (`ar-eg`).

amadib replied:
Another thing that will need to be updated along with this is the main [README](https://github.com/x1ao4/edition-manager-for-plex?tab=readme-ov-file#modules)

Entree3k replied:
> Also referred to as `locales`. Here's another [reference](https://en.wikipedia.org/wiki/Language_localisation).
> 
> Personally example much of my library includes Arabic that's either dubbed into both classical/modern standard arabic (`ar`) or egyptian arabic (`ar-eg`).

![image](https://github.com/x1ao4/edition-manager-for-plex/assets/28127566/776c83c2-9655-4ab1-b5ca-5a167b27ee5a)
My languages are based on language codes you would find in mkvtoolnix to define audio tracks

amadib replied:
I see. So would I be using [this designation](https://github.com/x1ao4/edition-manager-for-plex/pull/1/commits/ed376592824ea6d67901cf07780b492961403698#diff-fe786fec43e8e0969f38b72df1f80e9788f4865fdbc5d3b5ed5bbdf1691309c4R1886) going forward for any `ar-eg` files? 

<img src="https://github.com/x1ao4/edition-manager-for-plex/assets/326888/fa01784c-5805-42ab-8c4e-338a3e41332f" width="400" height="300"/>

Entree3k committed:
Update Language.py

Entree3k committed:
Update README.md

Entree3k committed:
Update Language.py

amadib commented:
@x1ao4 this would be a welcomed addition

x1ao4 commented:
> @x1ao4 this would be a welcomed addition

I've been a bit busy lately, but I'll review and consider this PR when I have some free time.

Entree3k committed:
Delete Dockerfile

Entree3k committed:
Delete start.sh

Entree3k committed:
Delete template directory

Entree3k committed:
Delete edition-manager-for-plex.py

Entree3k committed:
Delete emp-all.bat

Entree3k committed:
Delete emp-all.command

Entree3k committed:
Delete emp-new.command

Entree3k committed:
Delete emp-reset.bat

Entree3k committed:
Delete emp-new.bat

Entree3k committed:
Delete emp-reset.command

Entree3k committed:
Delete requirements.txt

Entree3k committed:
Delete modules/AudioCodec.py

Entree3k committed:
Delete modules directory

Entree3k committed:
Update config.ini

Entree3k committed:
Add files via upload

Entree3k committed:
Rename edition-manager-for-plex.py to edition-manager.py

Entree3k committed:
Create dockerfile

Entree3k committed:
Rename dockerfile to Dockerfile

Entree3k committed:
Delete compose/edition-manager-for-plex-reset directory

Entree3k committed:
Delete compose/edition-manager-for-plex directory

Entree3k committed:
Create compose.yml

Entree3k committed:
Update compose.yml

Entree3k committed:
Update Dockerfile

Entree3k committed:
Add files via upload

Entree3k committed:
Update edition-manager.py

Entree3k committed:
Update edition-manager.py

Entree3k committed:
Update config.ini

Entree3k committed:
Update compose.yml

Entree3k committed:
Update Dockerfile

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Update edition-manager.py

Entree3k committed:
Added new modules

Source, Size, FrameRate, Bitrate, AudioCodec and VideoCodec

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

amadib commented:
@x1ao4 just a friendly nudge

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Update config.ini

Entree3k committed:
Update edition-manager-gui.py

Entree3k committed:
Update requirements.txt

Entree3k committed:
Update edition-manager.py

Entree3k committed:
Update edition-manager.py

Entree3k committed:
Update Language.py

Entree3k committed:
Create Edition Manager Modules.md

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Update requirements.txt

Entree3k committed:
Removed - cache_ttl

Entree3k committed:
Update edition-manager-gui.py

Entree3k committed:
Create Edition Manager GUI.md

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Add files via upload

Entree3k committed:
Update Edition Manager GUI.md

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Delete compose.yml

Entree3k committed:
Delete Dockerfile

Entree3k committed:
Delete edition-manager-gui.bat

Entree3k committed:
Add files via upload

Entree3k committed:
Update config.ini

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Add files via upload

Entree3k committed:
Add files via upload

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Add files via upload

Complete module rework

Made them more efficient so they don't have to hit Plex for each module for each movie. Instead they will pull information from the movie and share it with each module that needs it.

Added

Writer (lists the first credited writer for the movie)
Short Film (lists movies that are < 40 min)

Entree3k committed:
Add files via upload

Entree3k committed:
Update Edition Manager Modules.md

Added Writer and Short Film

Entree3k committed:
Update Edition Manager GUI.md

Entree3k committed:
Update README.md

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Delete edition-manager-gui.py

Entree3k committed:
Delete edition-manager-gui.sh

Entree3k committed:
Delete edition-manager-gui.pyw

Entree3k committed:
Delete edition-manager.py

Entree3k committed:
Update config.ini

Entree3k committed:
Add files via upload

Entree3k committed:
Create file_delete.txt

Entree3k committed:
Add files via upload

Entree3k committed:
Delete assets/file_delete.txt

Entree3k committed:
Update messages.txt

Entree3k committed:
Add files via upload

Entree3k committed:
Update messages.txt

Entree3k committed:
Update README.md

Entree3k committed:
Add files via upload

Entree3k committed:
Add files via upload

Fixed: AttributeError: 'MainWindow' object has no attribute 'percent_lab' Traceback

Entree3k committed:
Update README.md

Entree3k committed:
Update Edition Manager GUI.md

Entree3k committed:
Fixed issue where it wasn't reading from config

Entree3k committed:
Added short code patterns for Cut editions

Entree3k committed:
Update config.ini

Entree3k committed:
Add files via upload
